agent/config.go

```diff
@@ -16,4 +16,5 @@ type Config struct {
 	MaxRetries           int
 	RetryInitialInterval time.Duration
 	StopOnSinkError      bool
+	TimerFn              func() func() int
```

Comment:
let's create a type for this function e.g. TimerFn instead of using `func()func()int` everywhere